L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryHexImp2.cc

```diff
@@ -314,16 +417,130 @@ getOrderedTriggerCellsFromModule( const unsigned module_id ) const
+                break;
+            default:
+                edm::LogError("HGCalTriggerGeometry") << "Unknown wafer neighbours for subdet "<<subdet<<"\n";
+                return geom_set();
+        } 
+    }
+    catch (const std::out_of_range& e) {
+        throw cms::Exception("BadGeometry")
+            << "HGCalTriggerGeometry: Neighbors are not defined for wafer " << wafer << " in subdetector " << subdet
+            << ". The wafer neighbor mapping should be modified. \n";
+    };
+    if(!surrounding_wafers)
+    {
+        throw cms::Exception("BadGeometry")
+            << "HGCalTriggerGeometry: Neighbors are not defined for wafer " << wafer << " in subdetector " << subdet
+            << ". The wafer neighbor mapping should be modified. \n";
+    }
+    // Find the types of the surrounding wafers
+    std::vector<int> types;
+    types.reserve(surrounding_wafers->size()+1); // includes the central wafer -> +1
+    types.emplace_back(wafer_type);
+    for(const auto w : *surrounding_wafers)
+    {
+        // if no neighbor, use the same type as the central one
+        // to create the wafer configuration
+        int wt = wafer_type;
+        if(w!=-1)
+        {
+            switch(subdet)
+            {
+                // HGCalDDDConstants::waferTypeT() returns 2=coarse, 1=fine
+                // HGCalDetId::waferType() returns -1=coarse, 1=fine
+                // Convert to HGCalDetId waferType
+                case ForwardSubdetector::HGCEE:
+                    wt = (es_info_.topo_ee->dddConstants().waferTypeT(w)==2?-1:1);
+                    break;
+                case ForwardSubdetector::HGCHEF:
+                    wt = (es_info_.topo_fh->dddConstants().waferTypeT(w)==2?-1:1);
+                    break;
+                default:
+                    // Should not arrive there. Subdet has been checked before
+                    return geom_set();
+            } 
+        }
+        types.emplace_back(wt);
+    }
+    // retrieve neighbors
+    unsigned trigger_cell_key = packTriggerCell(trigger_cell, types);
+    geom_set neighbor_detids;
+    try 
+    {
+        const auto& neighbors = trigger_cell_neighbors_.at(trigger_cell_key);
+        // create HGCalDetId of neighbors and check their validity
+        neighbor_detids.reserve(neighbors.size());
+        for(const auto& wafer_tc : neighbors)
+        {
+            unsigned neighbor_wafer = (wafer_tc.first==0 ? wafer : surrounding_wafers->at(wafer_tc.first-1));
+            int type = types.at(wafer_tc.first);
+            HGCalDetId neighbor_det_id((ForwardSubdetector)trigger_cell_det_id.subdetId(), trigger_cell_det_id.zside(), trigger_cell_det_id.layer(), type, neighbor_wafer, wafer_tc.second);
+            if(validTriggerCell(neighbor_det_id.rawId()))
+            {
+                neighbor_detids.emplace(neighbor_det_id.rawId());
+            }
+        }
+    }
+    catch (const std::out_of_range& e) {
```

Comment:
In most of the cases in this file `out_of_range` exceptions are caught for `unordered_map::at()`.
In these cases, instead of `at()` and `try/catch` I will use `find()` and check the returned iterator. Is it ok?